to_do_app/src/Reducers/todoReducer.js

```diff
@@ -0,0 +1,31 @@
+import { ADD_TODO, TOGGLE_TODO } from "../Actions/actionTypes";
+
+import { addTodo, toggleToDo } from "../Actions/actions";
+//do i need to import the action functions?
+
+//is there where i say what i want the initial state to be? and should it be a todo since this is what we are updating?
+const initialState = {
+  id: 0,
+  text: "",
+  completed: false
+};
+
+const todoReducer = (state = initialState, action) => {
+  //reducers update application state by action types?
```

Comment:
Action types are how the reducers decide what action they have to do to return the newState